File: node-neo4j-driver/insert_friends.js
const getNeo4jDriver = require("./yaml-config");
const driver = getNeo4jDriver("../config.yml");
const session = driver.session();

var insertQuery = 
  "UNWIND {pairs} as pair \
   MERGE (p1:Person {name:pair[0]}) \
   MERGE (p2:Person {name:pair[1]}) \
   MERGE (p1)-[:KNOWS]-(p2)";

var data = [["Jim","Mike"],["Jim","Billy"],["Anna","Jim"],
            ["Anna","Mike"],["Sally","Anna"],["Joe","Sally"],
            ["Joe","Bob"],["Bob","Sally"]];

session
    .run(insertQuery, {pairs: data})
    .then(function (result) {
        session.close();
        result.records.forEach(function (record) {
            console.log(record);
        });
        driver.close();
    })
    .catch(function (error) { 
        console.log(error);
    });

File: elixir-neo4j-driver/social/insert_friends.exs
defmodule Knows do
  def create(conn, pairs) do
    insert_query = """
    UNWIND {pairs} as pair
    MERGE (p1:Person {name:pair[0]})
    MERGE (p2:Person {name:pair[1]})
    MERGE (p1)-[:KNOWS]-(p2)
    """

    Bolt.Sips.query!(conn, insert_query, %{pairs: pairs}).results |> IO.inspect
  end
end

[[{'userName', username}, {'password', password}, {'host', host}, _ , _, _]] = :yamerl_constr.file("../../config.yml")
{:ok, _neo} = Bolt.Sips.start_link(url: "bolt://#{username}:#{password}@#{host}")
conn = Bolt.Sips.conn()

pairs = [["Jim", "Mike"], ["Jim", "Billy"], ["Anna", "Jim"],
        ["Anna", "Mike"], ["Sally", "Anna"], ["Joe", "Sally"],
        ["Joe", "Bob"], ["Bob", "Sally"]]

Knows.create(conn, pairs)
